src/openrct2/OpenRCT2.cpp

```diff
@@ -228,14 +228,27 @@ extern "C"
                 {
 #ifndef DISABLE_HTTP
                     // Download park and open it using its temporary filename
-                    char tmpPath[MAX_PATH];
-                    if (!http_download_park(gOpenRCT2StartupActionPath, tmpPath))
+                    http_response_t *response = http_download_park(gOpenRCT2StartupActionPath);
+                    if (response == NULL)
                     {
                         title_load();
                         break;
                     }
 
-                    parkLoaded = OpenRCT2::OpenParkAutoDetectFormat(tmpPath);
+					SDL_RWops *rw = SDL_RWFromConstMem(response->body, response->size - 1);
```

Comment:
That's on my list of what to do for this.

I've been looking over it, and rewriting `OpenParkAutoDetectFormat` to work the way it needs to for this would require a minor refactoring of the file loading code. That may cause the PR to take longer than initially expected, but I'll get this fixed before I remove the WIP tag.